src/table/BaseRecord.cc

```diff
@@ -13,6 +13,55 @@ namespace table {
 
 namespace {
 
+// A Schema Functor used to set floating point-fields to NaN and initialize variable-length arrays
+// using placement new.  All other fields are left alone, as they should already be zero.
+struct RecordInitializer {
+    template <typename T>
+    static void fill(T *element, int size) {}  // this matches all non-floating-point-element fields.
+
+    static void fill(float *element, int size) {
```

Comment:
For my enlightenment: this doesn't need `template <>` because they're in a `struct` rather than free functions?

Reply:
Because it's a function and not a class or struct, actually - because you can have function overloads that have nothing to do with templates, a fully-specialized "function template" is just a regular function overload, and the `template <>` syntax only applies to classes and structs.